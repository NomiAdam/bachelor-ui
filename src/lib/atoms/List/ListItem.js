// eslint-disable-next-line no-unused-vars
import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';

const ListItem = styled(
    ({ component, ...props }) => createElement(component, props),
)`
	max-width: 100%;
	padding: 10px 0;
	background-color: transparent;
	transition: .25s all ease;
	&:hover {
		background-color: ${ lightTheme.whiteBlue };
	}
`;

ListItem.propTypes = {
    /**
	 * Children nodes of ListItem component
	 */
    children: PropTypes.any,
    /**
	 * String representation of used ListItem component, "li", "div", ...
	 */
    component: PropTypes.string,
};

ListItem.defaultProps = {
    component: 'li',
};

export default ListItem;
