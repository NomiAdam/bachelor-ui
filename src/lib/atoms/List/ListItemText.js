import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { darkTheme } from '../../constants/theme';

export const StyledListItemText = styled.span`
	color: ${darkTheme.black};
	font-size: 16px;
`;

export const StyledListItemTextDiv = styled.div`
	width: 100%;
	height: 100%;
	text-align: ${props => props.align};
`;

const ListItemText = ({align, primary}) => (
	<StyledListItemTextDiv align={align}>
		<StyledListItemText>
			{primary}
		</StyledListItemText>
	</StyledListItemTextDiv>
);


ListItemText.propTypes = {
	/**
	 * String representing alignment of ListItemText text
	 */
	align: PropTypes.string,
	/**
	 * String representing primary value
	 */
	primary: PropTypes.string
};

ListItemText.defaultProps = {
    align: 'left',
    primary: '',
};

export default ListItemText;
