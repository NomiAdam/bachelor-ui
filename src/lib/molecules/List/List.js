import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const StyledList = styled.ul`
	width: 100%;
	height: 100%;
	padding: 0;
	list-style-type: none;
  margin: 0;
`;

const List = ({ children }) => (<StyledList>{children}</StyledList>);

List.propTypes = {
    /**
	 * Children nodes of List component, ListItem component is expected
	 */
    children: PropTypes.any,
};

export default List;
