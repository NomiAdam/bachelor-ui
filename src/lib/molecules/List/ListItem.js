// eslint-disable-next-line no-unused-vars
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

const StyleListItem = styled.li`
	max-width: 100%;
	padding: 10px 0;
	background-color: transparent;
	transition: .25s all ease;
  ${ ({ disableBorder, ...props }) => (disableBorder ? (
        ''
    ) : (
        `border-bottom: 1px solid ${ resolveTheme(TYPES.COLOR)(props) };`
    ))
}
	&:hover {
	  cursor: ${ ({ pointer }) => (pointer ? 'pointer' : 'initial') };
		background-color: ${
    ({ disableHover, ...props }) => (disableHover ? 'transparent' : resolveTheme(TYPES.BACKGROUND_COLOR_HOVER)(props))
};
}
`;

const ListItem = ({ children, ...props }) => (
    <StyleListItem { ...props }>
        {children}
    </StyleListItem>
);

ListItem.propTypes = {
    /**
   * Children nodes of ListItem component
   */
    children: PropTypes.any,
    /**
   * Whether to show pointer
   */
    pointer: PropTypes.bool,
    /**
   * Whether to show border
   */
    disableBorder: PropTypes.bool,
    /**
   * Whether to disable hover effect
   */
    disableHover: PropTypes.bool,
};

export default ListItem;
