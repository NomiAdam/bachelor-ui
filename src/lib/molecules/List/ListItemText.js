import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

export const StyledListItemText = styled.span`
	color: ${ resolveTheme(TYPES.COLOR) };
	font-size: 16px;
`;

export const StyledListItemTextDiv = styled.div`
	width: 100%;
	height: 100%;
	text-align: ${ ({ align }) => align };
`;

const ListItemText = ({ align, primary }) => (
    <StyledListItemTextDiv align={ align }>
        <StyledListItemText>
            {primary}
        </StyledListItemText>
    </StyledListItemTextDiv>
);

ListItemText.propTypes = {
    /**
	 * String representing alignment of ListItemText text
	 */
    align: PropTypes.string,
    /**
	 * String representing primary value
	 */
    primary: PropTypes.string,
};

ListItemText.defaultProps = {
    align: 'left',
    primary: '',
};

export default ListItemText;
