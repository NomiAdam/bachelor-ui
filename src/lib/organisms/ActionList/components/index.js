import ActionListItem from './ActionListItem';

export default ActionListItem;
