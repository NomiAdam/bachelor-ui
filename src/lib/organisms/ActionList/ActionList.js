import React, { PureComponent } from 'react';
import { map, prop, propOr } from 'ramda';
import PropTypes from 'prop-types';
import List from '../../atoms/List/index';
import ActionListItem from './components/index';

const getIdFromProp = prop('id');
const getActiveProp = propOr(false, 'active');
const renderActionListItem = (handleClick, handleIconClick, isAction) => props => (
    <ActionListItem
        { ...props }
        isAction={ isAction }
        key={ getIdFromProp(props) }
        isIconActive={ getActiveProp(props) }
        handleClick={ handleClick(getIdFromProp(props)) }
        handleIconClick={ handleIconClick(getIdFromProp(props), getActiveProp(props)) }
    />
);
const renderActionList = (handleClick, handleIconClick, isAction) => map(
    renderActionListItem(handleClick, handleIconClick, isAction),
);
class ActionList extends PureComponent {
    handleItemClick = index => () => {
        const { handleItemClick } = this.props;
        handleItemClick(index);
    };

    handleIconClick = (index, isActive) => () => {
        const { handleIconClick } = this.props;
        handleIconClick(index, isActive);
    };

    render() {
        const { listItems, isAction } = this.props;
        return (
            <List>
                {renderActionList(
                    this.handleItemClick,
                    this.handleIconClick,
                    isAction,
                )(listItems)}
            </List>
        );
    }
}

ActionList.propTypes = {
    /**
     * Array of objects, format below is expected,
     * [ { id, date, header, subHeader, active }, ... ]
     */
    listItems: PropTypes.array,
    /**
     * Function to be called upon link click, note that function is called with, item ID as param
     */
    handleItemClick: PropTypes.func,
    /**
     * Function to be called upon icon click,
     */
    handleIconClick: PropTypes.func,
    /**
     * Boolean indicating whether we use list for actions,
     * Is case of true pointer css will be set and so on
     */
    isAction: PropTypes.bool,
};

ActionList.defaultProps = {
    listItems: [],
    isAction: false,
};

export default ActionList;
