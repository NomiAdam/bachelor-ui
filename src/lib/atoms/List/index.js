import List from './List';
import ListItem from './ListItem';
import ListItemText from './ListItemText';

export {
	ListItem,
	ListItemText
};

export default List;
