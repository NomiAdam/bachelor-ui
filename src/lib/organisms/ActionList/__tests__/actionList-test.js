/* eslint-disable no-undef */
import React from 'react';
import List from '../../../atoms/List/index';
import ActionList from '../ActionList';
import {shallow} from 'enzyme';

describe('ActionList', () => {
    it('should be defined', () => {
        expect(ActionList).toBeDefined();
    });
    it('should be match the snapshot', () => {
        expect(shallow(<ActionList/>)).toMatchSnapshot();
    });
    it('should be of type List', () => {
        expect(shallow(<ActionList/>).type()).toBe(List);
    })
});
