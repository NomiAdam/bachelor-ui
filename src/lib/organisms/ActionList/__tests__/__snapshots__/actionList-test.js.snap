// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionList should be match the snapshot 1`] = `<List />`;
